internal/providers/packet/packet.go

```diff
@@ -48,3 +62,107 @@ func FetchConfig(f resource.Fetcher) (types.Config, report.Report, error) {
 
 	return util.ParseConfig(f.Logger, data)
 }
+
+// FetchStatus is invoked by oem.Status() when igintion fails
+// POSTS if s==fail, a message to Packet's provision setup Timeline
+func FetchStatus(f resource.Fetcher, s string) error {
+
+	type Message struct {
+		Message string `json:"message"`
+	}
+
+	// fetch JSON from https://metadata.packet.net/metadata
+	data, err := f.FetchToBuffer(MetadataUrl, resource.FetchOptions{
+		Headers: nil,
+	})
+	if data == nil {
+		log.Fatalf("data is nil")
+	}
+
+	if err != nil {
+		fmt.Println("Error: ", err)
+	}
+
+	type Metadata struct {
+		PhoneHomeURL string `json:"phone_home_url"`
+	}
+
+	metadata := Metadata{}
+
+	err = json.Unmarshal(data, &metadata)
+	if err != nil {
+		fmt.Println("Error: ", err)
+	}
+
+	phonehomeURL := metadata.PhoneHomeURL
+	// to get phonehome IP
+	phonehomeURL = strings.TrimSuffix(phonehomeURL, "/phone-home")
+
+	// POST Message to phonehome IP
+	postMessageURL := phonehomeURL + "/events"
+
+	// Place message into report struct
+	if s == "fail" {
+		m := Message{
+			Message: "Ignition Config Invalid",
+		}
+		// Marshall Message
+		messageJSON, err := json.Marshal(m)
+		if err != nil {
+			fmt.Println("Error: ", err)
+		} else {
+			fmt.Println("successfully encoded message, here it is: %s\n", string(messageJSON))
+		}
+		// POST Message
+		err = postMessage(messageJSON, postMessageURL)
+		if err != nil {
+			fmt.Println("Error: ", err)
+		}
+
+	} else {
+
+		// any string other than "fail" in vairable s:
+		fmt.Println("IGNTION STATUS PASS")
+
+		m := Message{
+			Message: "Ignition Config valid",
+		}
+
+		// Marshall Message
+		messageJSON, err := json.Marshal(m)
+		if err != nil {
+			fmt.Println("Error: ", err)
+		} else {
+			fmt.Println("successfully encoded message: %s", string(messageJSON))
+		}
+		// POST Message
+		err = postMessage(messageJSON, postMessageURL)
+		if err != nil {
+			fmt.Println("Error: ", err)
+		}
+
+	}
+
+	return err
+}
+
+// postMessage(): POST message in JSON to API URL
+func postMessage(message []byte, url string) error {
+
+	postReq, err := http.NewRequest("POST", url, bytes.NewBuffer(message))
+	if err != nil {
+		fmt.Println("Error: ", err)
+	}
+	postReq.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
```

Comment:
@ajeddeloh: would it make sense to make a `GetHttpClient` function of the `Fetcher` object and/or make a `Post` call? That way we can get clients with the correct `Transport` & `http.Client` customizations easier?

Reply:
hmmm actually I don't think so. Ignition could have failed because of wonky timeouts/CAs. We shouldn't need those customizations here, especially since this is packet specific.